### DISCUSSION TITLE
Prima port

### DISCUSSION THREAD

**Discussion by kakarot-dev** [prisma/prisma#8908] [Q&A] [answered]
Does prisma open a port by itself?

**Comment by ryands17** [comment, marked as answer]
Have a look at this discussion where this is answered: https://github.com/prisma/prisma/discussions/8754

**Reply by kakarot-dev** [reply]
Which port does it open? Is there a way to set it?

**Reply by kakarot-dev** [reply]
Or do I have to use node api?

**Reply by ryands17** [reply]
It's written in the answer that it opens port `36917` to communicate with the Query Engine. Using Node API removes the need for this server and makes your queries faster.

**Reply by pantharshit00** [reply]
Port is random, it uses any free port. If you don't want a port open, try the node api preview. We will make Node API the default in the near future: https://github.com/prisma/prisma/issues/8731

**Reply by kakarot-dev** [reply]
I want to host my app on docker, so is there any need to explicitly expose a port for prisma then?

**Reply by ryands17** [reply]
No that's not needed.

**Reply by pantharshit00** [reply]
Communication between the engine and the client happens on the same machine so no need for that.

**Reply by kakarot-dev** [reply]
Tyyyy so much! I am using prisma cuz of all the features it has, keep up the good work!
And hopefully cockroachdb will be supported soon :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
